sale_purchase_order_line/models/product_template.py

```diff
@@ -0,0 +1,16 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class ProductTemplate(models.Model):
+    _name = "product.template"
+    _inherit = ["product.template", "sale.purchase.order.line.mixin"]
+
+    def _compute_domain(self):
+        return [('product_id', 'in', self.product_variant_ids.ids)]
+        #return [('product_id.product_tmpl_id', '=', self.id)]
+    
+    def _compute_context(self):
+        return {'hide_product': True}
```

Comment:
Use a more detailed context variable name, because any other addon could add to context a variable with a very close name. E.g. we could include in variable name addon name.

And we should add new desired context to existing one, so we shouldn't set in, but combine with existining one (even if we know that is already empty). Not tested, please try:

```suggestion
        ret = super()._compute_context()
        ret.update({'hide_product': True})
        return ret
```